scripts/scripts.js

```diff
@@ -132,15 +133,228 @@ export function decorateMain(main) {
+    window.alloy('sendEvent', {
+      xdm: {
+        eventType: 'decisioning.propositionDisplay',
+        _experience: {
+          decisioning: { propositions },
+        },
+      },
+    });
+  });
+}
+
+async function sendAnalyticsEvent(xdmData) {
+  // eslint-disable-next-line no-undef
+  if (!alloy) {
+    return Promise.resolve();
+  }
+  // eslint-disable-next-line no-undef
+  return alloy('sendEvent', {
+    documentUnloading: true,
+    xdm: xdmData,
+  });
+}
+
+/**
+ * Basic tracking for page views with alloy
+ * @param document
+ * @param additionalXdmFields
+ * @returns {Promise<*>}
+ */
+// eslint-disable-next-line no-unused-vars
+async function analyticsTrackPageViews(document, additionalXdmFields = {}) {
+  const xdmData = {
+    eventType: 'web.webpagedetails.pageViews',
+    web: {
+      webPageDetails: {
+        pageViews: {
+          value: 1,
+        },
+        name: document.title,
+      },
+    },
+  };
+  sendAnalyticsEvent(xdmData);
+}
+
+function initializeTargetAnalytics() {
+  const alloyLoadedPromise = initWebSDK(`${window.hlx.codeBasePath}/scripts/alloy.min.js`);
+
+  if (getMetadata('target')) {
+    alloyLoadedPromise.then(() => {
+      fetch(`${getHostUrl()}/websdkconfig.json`)
+        .then((response) => response.json())
+        .then((configData) => {
+          // eslint-disable-next-line no-undef
+          alloy('configure', {
+            datastreamId: configData.data[0].DatastreamId,
+            orgId: configData.data[0].OrgId,
+          });
+          getAndApplyRenderDecisions();
+          analyticsTrackPageViews(document);
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.error('Error:', error);
+        });
+    });
+  }
+}
+
+/**
+ * Returns script that initializes a queue for each alloy instance,
+ * in order to be ready to receive events before the alloy library is loaded
+ * Documentation
+ * https://experienceleague.adobe.com/docs/experience-platform/edge/fundamentals/installing-the-sdk.html?lang=en#adding-the-code
+ * @type {string}
+ */
+function getAlloyInitScript() {
+  return `!function(n,o){o.forEach(function(o){n[o]||((n.__alloyNS=n.__alloyNS||[]).push(o),n[o]=
+  function(){var u=arguments;return new Promise(function(i,l){n[o].q.push([i,l,u])})},n[o].q=[])})}(window,["alloy"]);`;
+}
+
+/**
+ * Create inline script
+ * @param document
+ * @param element where to create the script element
+ * @param innerHTML the script
+ * @param type the type of the script element
+ * @returns {HTMLScriptElement}
+ */
+function createInlineScript(document, element, innerHTML, type) {
+  const script = document.createElement('script');
+  script.type = type;
+  script.innerHTML = innerHTML;
+  element.appendChild(script);
+  return script;
+}
+
 /**
  * Loads everything needed to get to LCP.
  * @param {Element} doc The container element
  */
 async function loadEager(doc) {
+  document.body.classList.add('visibility-hidden');
```

Comment:
The body should be hidden by default in the boilerplate, so not sure why you need the additional visibility CSS classes throughout here.

Reply:
@ramboz yes, I will handle with 'appear' class which is already present in the code here https://github.com/aemsites/icicidirect/pull/280/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cR359